### ISSUE TITLE
http_upload: File upload with Metronome

### ISSUE THREAD

**Issue by M5oul** [YunoHost/issues#1278]
In order to upload files with Metronome, aka `http_upload`, the instance needs a `upload.main_domain.tld` domain.

To do so, the admin should add this domain name, a DNS redirection (if a wirdcard is not already present) as well as generating the LE certificate (the upload did works without it).

The issue could be that the configuration file created for this domain name: `/etc/metronome/conf.d/upload.main_domain.tld.cfg.lua` conflicts with the upload component in main Metronome configuration file preventing Metronome to restart (There is no log saying what is happening).

We should find a way to not generate this configuration file, even for `regen-conf` actions (automatic, manual).

It must be documented here https://yunohost.org/#/XMPP and here https://yunohost.org/#/dns_config_fr

---

Cf https://forum.yunohost.org/t/metronome-envoie-de-fichier/6607/3

**Comment by e-jim** [comment]
I would add that for it to work on most client one needs to have encryption on the http port as noted by kcalsnur https://forum.yunohost.org/t/cant-upload-file-via-xmpp-server/7164/3

He appended this to /etc/metronome/metronome.cf.lua:

`Host "*"
       interfaces = { "*" }
       ssl = { key = "/etc/yunohost/certs/upload.maillard.im/key.pem", certificate = "/etc/yunohost/certs/upload.maillard.im/crt.pem" }
      http_ports = {}
      https_ports = { 5290 }
`

I tested, and it works when dropping the host and interface directives and editing the HTTP server section this way:

```
-- HTTP server                                                                                                                                                                                                    
ssl = { key = "/etc/yunohost/certs/upload.chezj.im/key.pem", certificate = "/etc/yunohost/certs/upload.chezj.im/crt.pem" }                                                                                        
http_ports = {}
https_ports = { 5290 }
```

**Comment by decentral1se** [comment]
I struggled to set this up and still have not managed it. I had to open port 5290 (we're missing that from the DNS and ports documentation) and tried generating a certificate for my `upload.myyunohostdomain.org` but that caused metronome to complain something about a conflicting virtual host and a component. I'll try to continue to get this working soon and update the documentation as my understanding improves ... any help appreciated (if any of this even makes sense ;)).

**Comment by bjarkan** [comment]
[metronome.cfg (1).lua.txt](https://github.com/YunoHost/issues/files/3494325/metronome.cfg.1.lua.txt)
[upload.conf.txt](https://github.com/YunoHost/issues/files/3494327/upload.conf.txt)
[share.php.txt](https://github.com/YunoHost/issues/files/3494335/share.php.txt)



I already cover that in my blog https://translate.google.es/translate?um=1&ie=UTF-8&hl=en&client=tw-ob&sl=es&tl=en&u=https%3A%2F%2Fmiradordemontepinar.net%2Fblog%2Fblog%2F2019%2F08%2F08%2Fyunohost-selfhosting-hecho-facil-parte-cinco-aplicaciones%2F
in the metronome section...

TLDR;
Basically add this to /etc/metronome/metronome.cfg.lua
......................................
---Set up a HTTP Upload service
Component "upload.vuestrodominio.net" "http_upload_external"
	name = "vuestrodominio.net Sharing Service"
	interfaces = {"*"}
	http_file_external_url = "https://upload.vuestrodominio.net/"
	http_file_secret = "el secreto que yo elija"
	http_file_size_limit = 100*1024*1024
	http_file_external_delete_url = "https://upload.vuestrodominio.net/"
	http_file_delete_secret = "el secreto para borrar que yo quiera"
	http_file_expire_after = 90*60*60*24
..................................

for godness sake "vuestrodominio.net" whatever dns it is but yours...  "el secreto que yo elija" "el secreto para borrar que quiera" 2 secret passwords...

in /var/www/upload put this file https://github.com/maranda/metronome/blob/master/plugins/http_upload_external/resources/share.php 
PLEASE use the same "secrets" in both files....

in  /etc/nginx/conf.d/upload.yourdomain.net.d/upload.conf JUST THIS
.............................
rewrite ^.*$ /share.php$fastcgi_script_name;
    client_max_body_size 60m;

    root /var/www/upload/;
    index index.php share.php;

    try_files $uri $uri/ /share.php;

    location ~ \.php {
    fastcgi_pass unix:/var/run/php/php7.0-fpm.sock;
    include fastcgi_params;
    fastcgi_split_path_info ^(.+?\.php)(/.*)$;
    fastcgi_param HTTPS on;
    fastcgi_param  PATH_INFO $fastcgi_path_info;
    fastcgi_param PATH_TRANSLATED $request_filename;
    }
......................................

in the bash # systemctl restart nginx and systemctl restart metronome

**Comment by pitchum** [comment]
I think we can close this issue and maybe add "milestone 3.8.x"
